src/sql-parser/src/parser.rs

```diff
@@ -1891,19 +1891,31 @@ impl<'a> Parser<'a> {
             Keyword::Kafka => {
                 self.expect_keyword(BROKER)?;
                 let broker = self.parse_literal_string()?;
-                let with_options = self.parse_opt_with_options()?;
-                CreateConnector::Kafka {
-                    broker,
-                    with_options,
-                }
+                self.expect_keyword(SECURITY)?;
+                let security = self.parse_kafka_security()?;
+                CreateConnector::Kafka { broker, security }
             }
             Keyword::Confluent => {
                 self.expect_keywords(&[SCHEMA, REGISTRY])?;
                 let registry = self.parse_literal_string()?;
-                let with_options = self.parse_opt_with_options()?;
+                let (mut username, mut password) = (None, None);
+                while let Ok(keyword) = self.expect_one_of_keywords(&[USERNAME, PASSWORD]) {
+                    match keyword {
+                        Keyword::Username => {
+                            let _ = self.consume_token(&Token::Eq);
+                            username = Some(self.parse_literal_string()?);
+                        }
+                        Keyword::Password => {
+                            let _ = self.consume_token(&Token::Eq);
+                            password = Some(self.parse_object_name()?);
```

Comment:
I don't think `parse_object_name` is the right call here; maybe `parse_identifier` or we can force all of these values to be delimited as strings with `parse_literal_string`

Reply:
Oh, do these all maybe want to be `parse_with_option_value`?

Reply:
Historically these were string values but that is not going to be true in a platform context, instead these are names of `Secret` items in the catalog which will be looked up the same way `Connector` names are for sources, so I thought it'd be correct to use the same parsing pattern as well.

Reply:
Ah, I didn't realize that. Good call.